### PR TITLE
Fix LAMMPS pair potential

### LINKED ISSUE #135: LAMMPS miscellaneous fixes
* [x] The potential table needs to use r^2 rather than r. Otherwise, LAMMPS does cubic spline interpolation, which can produce very wrong results when the potential becomes large.
* [x] The cutoff may be off by one r value when potentials have cutoffs set at their minimum (e.g., WCA). We should scan these cutoffs when deciding where to cut the potential.
* [x] The default delay for rebuilding the neighbor list is 10, but it should be 0.
* [x] The `EnsembleAnalyzer` should return a 2d box for 2d simulations. It always returns a `TriclinicBox`.

### PR DESCRIPTION
**Pull request by mphoward**
Fixes #135

### CODE REVIEW

File: relentless/simulate/lammps.py [hunk 1]
```diff
@@ -951,28 +955,42 @@ def pair_map(sim, pair):
...
                         )
 
                     # find r where potential and force are zero
-                    nonzero_r = numpy.flatnonzero(
-                        numpy.logical_and(~numpy.isclose(u, 0), ~numpy.isclose(f, 0))
-                    )
-                    if len(nonzero_r) > 1:
-                        # cutoff at last nonzero r (cannot be first r)
-                        rcut[(i, j)] = r[nonzero_r[-1]]
+                    all_rmax = [
```
Comment (clpetix) [inline relentless/simulate/lammps.py:976]: all_rmax's type is relentless.model.variable.Product

File: relentless/simulate/lammps.py [hunk 2]
```diff
@@ -951,28 +955,42 @@ def pair_map(sim, pair):
...
+                        pair_pot.coeff[i, j]["rmax"]
+                        for pair_pot in sim.potentials.pair.potentials
+                    ]
+                    if None not in all_rmax:
+                        # use rmax if set for all potentials
+                        rcut[(i, j)] = min(max(all_rmax), sim.potentials.pair.stop)
```
Comment (clpetix) [inline relentless/simulate/lammps.py:982]: Can't take the min of float & product
`TypeError: '<' not supported between instances of 'float' and 'Product'`

File: relentless/simulate/lammps.py (hunk 1 above)
Comment (mphoward) [reply relentless/simulate/lammps.py:976]: We need to make sure we evaluate these if it's a Variable.

**Comment by mphoward** [comment]
The doc builds are failing, I think because RTD is using a newer version of sphinx than we do. It may not be possible to actually pin the sphinx version on RTD, so I will updated our requirements and see if I can reproduce & fix.

**Review by clpetix — APPROVED** [review]
Looks good!

**Pull request merged**